pwem/protocols/protocol_boxsize_parameters.py

```diff
@@ -0,0 +1,299 @@
+# * it under the terms of the GNU General Public License as published by
+# * the Free Software Foundation; either version 3 of the License, or
+# * (at your option) any later version.
+# *
+# * This program is distributed in the hope that it will be useful,
+# * but WITHOUT ANY WARRANTY; without even the implied warranty of
+# * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# * GNU General Public License for more details.
+# *
+# * You should have received a copy of the GNU General Public License
+# * along with this program; if not, write to the Free Software
+# * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+# * 02111-1307  USA
+# *
+# *  All comments concerning this program package may be sent to the
+# *  e-mail address 'scipion@cnb.csic.es'
+# *
+# **************************************************************************
+import pyworkflow.protocol.params as params
+from pwem.protocols import EMProtocol
+from pyworkflow.object import Integer
+from collections import OrderedDict
+
+class ProtBoxSizeParameters(EMProtocol):
+    """
+    Protocol to make mathematical operations on particle picking boxsize
+    """
+
+    _label = 'box size related parameters'
+    listExpressions = []
+
+    def __init__(self, **args):
+        EMProtocol.__init__(self, **args)
+
+    def _defineParams(self, form):
+        """
+        Defines the parameters the protocol form will show and its behaviour
+        :param form:
+        """
+        form.addSection(label='Input')
+
+        form.addParam('inputMicrographs', params.PointerParam,
+                      pointerClass='SetOfMicrographs',
+                      important=True,
+                      label="Input micrographs",
+                      help='Select the SetOfMicrographs from where we extract the sampling rate')
+
+        form.addParam('boxSize', params.IntParam,
+                      label='Particle box size (px)',
+                      allowsPointers=True,
+                      important=True,
+                      help='This is size of the boxed particles (in pixels)')
+
+        # Extraction box size
+        form.addParam('boolExtractPartBx', params.BooleanParam, default=False,
+                      label='Extract particles boxsize?',
+                      help='Select yes if you want to generate these parameters.')
+        groupEx = form.addGroup('Extract particles boxsize',
+                                condition="boolExtractPartBx==%d" % True)
+        groupEx.addParam('factorExtractPartBx', params.FloatParam, default=1.5,
+                         label='Factor to multiply the box size (px)',
+                         condition="boolExtractPartBx==%d" % True,
+                         help='Extraction box size (px) = picking box size (px) * *factor*')
+
+        # Gautomatch picking parameters
+        form.addParam('boolGautomatchParams', params.BooleanParam, default=False,
```

Comment:
Hi @danielmarchan3 , why not generating all possible outputs without asking. I'd remove the booleans.
What do you think?